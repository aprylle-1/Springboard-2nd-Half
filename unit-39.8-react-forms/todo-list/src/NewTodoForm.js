import React, { useState } from "react";
import { v4 as uuid } from 'uuid';
function NewTodoForm ({addTodo}) { 

    const [dataForm, setDataForm] = useState({task : ""})

    function handleFormChange (e) {
        const { name, value } = e.target
        setDataForm(data => {
            return {
                ...data,
                [name] : value
            }
        })
    }

    function handleAddTodo (e) {
        e.preventDefault()
        addTodo({
            task : dataForm.task,
            key : uuid()
        })
        setDataForm({task : ""})
    }
    return (
        <form onSubmit={handleAddTodo} className="NewTodoForm">
            <label htmlFor="task">Task</label>
            <input
            onChange={handleFormChange}
            name = "task"
            id = "task"
            value = {dataForm.task}
            />
            <button>Add todo</button>
        </form>
    )
}

export default NewTodoForm