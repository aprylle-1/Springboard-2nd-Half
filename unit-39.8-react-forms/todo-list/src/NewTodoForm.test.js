import { render, screen } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

test('renders without crashing', () => {
  render(<NewTodoForm />);
});

test('matches screenshot', () => {
    const { asFragment } = render(<NewTodoForm/>)
    expect(asFragment).toMatchSnapshot()
  })
  
