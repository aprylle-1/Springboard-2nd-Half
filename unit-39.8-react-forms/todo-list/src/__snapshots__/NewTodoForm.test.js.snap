// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches screenshot 1`] = `[Function]`;
